server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java

```diff
@@ -330,7 +400,7 @@ static boolean persistentTasksChanged(final ClusterChangedEvent event) {
     }
 
     /** Returns true if the task is not assigned or is assigned to a non-existing node */
-    public static boolean needsReassignment(final Assignment assignment, final DiscoveryNodes nodes) {
+    public static boolean isAssignedToValidNode(final Assignment assignment, final DiscoveryNodes nodes) {
```

Comment:
how about `requiresAssignment`? (is assigned suggest it is but we also consider unassigned tasks)